core/src/main/java/org/infinispan/interceptors/distribution/L1NonTxInterceptor.java

```diff
@@ -301,11 +317,15 @@ private BasicInvocationStage handleDataWriteCommand(InvocationContext ctx, DataW
       });
    }
 
-   private BasicInvocationStage removeFromLocalL1(InvocationContext ctx, DataWriteCommand command, Object returnValue)
-         throws Throwable {
+   private BasicInvocationStage removeFromLocalL1(InvocationContext ctx, DataWriteCommand command, Object returnValue) {
       if (ctx.isOriginLocal() && !cdl.localNodeIsOwner(command.getKey())) {
+         CompletableFuture<?> pendingAcks = commandAckCollector.getCollectorCompletableFuture(command.getCommandInvocationId(), false);
```

Comment:
I thought you had already waited for the same `CompletableFuture` in `TriangleInterceptor`, but I see now I was confusing `TriangleInterceptor` and `TriangleDistributionInterceptor`. WDYT about renaming `TriangleInterceptor` to `TriangleAckInterceptor`?